lfric_macros/check_macro_chains.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+##############################################################################
+# (c) Crown copyright 2025 Met Office. All rights reserved.
+# The file LICENCE, distributed with this code, contains details of the terms
+# under which the code may be used.
+##############################################################################
+"""
+Script to check that all version.py macro chains appear valid and that the final
+after tag of the chain matches metadata import statement for each rose app.
+Only intended to be run from rose-stem
+"""
+
+import os
+import re
+import shutil
+import subprocess
+import sys
+from apply_macros import ApplyMacros
+
+
+def run_command(command):
+    """
+    Run a subprocess command and return the result object
+    Inputs:
+        - command, str with command to run
+    Outputs:
+        - result object from subprocess.run
+    """
+    return subprocess.run(
+        command.split(),
+        capture_output=True,
+        text=True,
+        timeout=120,
+        shell=False,
+        check=False,
+    )
```

Comment:
Noticed you have duplicated functions across multiple files, is it possible to just import the one from apply_macros? I don't think you are doing anything different between each.

```suggestion
from apply_macros import ApplyMacros, run_command
```